src/parse.rs

```diff
@@ -775,6 +776,24 @@ impl fmt::Show for Atom {
     }
 }
 
+impl fmt::String for Atom {
```

Comment:
Yeah, it's the same code as `fmt::Show`. Unfortunately, we need a `fmt::String` implementation because it provides `.to_string()` which is used elsewhere. Maybe if we could change the code such that the `fmt::String` implementation calls the `fmt` function of the `fmt::Show` implementation?